x-pack/platform/plugins/shared/streams/server/lib/streams/helpers/migrate_on_read.test.ts

```diff
@@ -0,0 +1,221 @@
+    ...overrides,
+  };
+}
+
+describe('migrateOnRead', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    mockIsNeverCondition.mockReturnValue(false);
+    mockStreamsAsserts.mockImplementation(() => {});
+  });
+
+  describe('routing status migration', () => {
+    describe('when routing rules need status migration', () => {
+      it('should add enabled status to rules without status', () => {
+        const definition = createCompleteWiredStreamDefinition({
+          routing: [createRoutingRule()],
+        });
+
+        const result = migrateOnRead(definition);
+        const routing = getRoutingFromResult(result);
+
+        expect(routing).toHaveLength(1);
+        expect(routing[0].status).toBe('enabled');
+        expect(mockStreamsAsserts).toHaveBeenCalled();
+      });
+
+      it('should migrate never conditions to disabled status', () => {
+        mockIsNeverCondition.mockReturnValue(true);
+
+        const definition = createCompleteWiredStreamDefinition({
+          routing: [createRoutingRule({ where: { never: {} } })],
+        });
+
+        const result = migrateOnRead(definition);
+        const routing = getRoutingFromResult(result);
+
+        expect(routing).toHaveLength(1);
+        expect(routing[0].status).toBe('disabled');
+        expect(routing[0].where).toEqual({ always: {} });
+        expect(mockIsNeverCondition).toHaveBeenCalledWith({ never: {} });
+        expect(mockStreamsAsserts).toHaveBeenCalled();
+      });
+
+      it('should migrate multiple routing rules with mixed status', () => {
+        mockIsNeverCondition
+          .mockReturnValueOnce(false) // First rule
+          .mockReturnValueOnce(true) // Second rule
+          .mockReturnValueOnce(false); // Third rule
+
+        const definition = createCompleteWiredStreamDefinition({
+          routing: [
+            createRoutingRule({
+              destination: 'test.stream.1',
+              where: { field: 'service.name', eq: 'test1' },
+            }),
+            createRoutingRule({ destination: 'test.stream.2', where: { never: {} } }),
+            createRoutingRule({
+              destination: 'test.stream.3',
+              where: { field: 'service.name', eq: 'test3' },
+              status: 'enabled',
+            }),
+          ],
+        });
+
+        const result = migrateOnRead(definition);
+        const routing = getRoutingFromResult(result);
+
+        expect(routing).toHaveLength(3);
+        expect(routing[0].status).toBe('enabled');
+        expect(routing[0].where).toEqual({ field: 'service.name', eq: 'test1' });
+        expect(routing[1].status).toBe('disabled');
+        expect(routing[1].where).toEqual({ always: {} });
+        expect(routing[2].status).toBe('enabled');
+        expect(routing[2].where).toEqual({ field: 'service.name', eq: 'test3' });
+        expect(mockStreamsAsserts).toHaveBeenCalled();
+      });
+    });
+
+    describe('when routing rules already have status', () => {
```

Comment:
Tests below:
`'should rename unwired to classic'`
`'should add description if missing'`
...and others

shouldn't be under test suite `'routing status migration'`. Ideally each migration case should have its own standalone suite.

As a stretch, if the current implementation allows, a structure could be introduced similar to how Saved Objects [migrations](https://github.com/elastic/kibana/tree/main/x-pack/solutions/observability/plugins/synthetics/server/saved_objects/migrations/monitors) are handled in a separate migrations directory for example, so that each migration lives in it's own file and have its own tests.

Reply:
Makes sense, for this PR, it'd a too much of a stretch.

Are the last two tests: `'should add description if missing'` and `'should rename unwired to classic'` related to `describe('routing status migration' ...`? If not, I guess they could have their own describe blocks.